lib/appcanary/http.rb

```diff
@@ -0,0 +1,60 @@
+require "net/http"
+require "net/http/post/multipart"
+require "json"
+
+module Appcanary
+  class ServiceError < RuntimeError
+  end
+
+  class << self
+    def url_for endpoint
+      case endpoint
+      when :monitors
+        URI.parse("#{config[:base_uri]}/monitors/#{config[:name]}")
+      when :check
+        URI.parse("#{config[:base_uri]}/check")
+      end
+    end
+
+    def try_request_with(request_type, endpoint)
```

Comment:
why not make this method agnostic as to the kind of file being sent? i.e. have ship_gemfile care about what a `Bundler.default_lockfile` is; thereby, try_request_with just takes in a filename.